synapse/logging/_terse_json.py

```diff
@@ -18,32 +18,49 @@
 """
 import json
 import logging
-from typing import Optional
 
 _encoder = json.JSONEncoder(ensure_ascii=False, separators=(",", ":"))
 
+# The properties of a standard LogRecord.
+_LOG_RECORD_ATTRIBUTES = {
+    "args",
+    "asctime",
+    "created",
+    "exc_info",
+    # exc_text isn't a public attribute, but is used to cache the result of formatException.
+    "exc_text",
+    "filename",
+    "funcName",
+    "levelname",
+    "levelno",
+    "lineno",
+    "message",
+    "module",
+    "msecs",
+    "msg",
+    "name",
+    "pathname",
+    "process",
+    "processName",
+    "relativeCreated",
+    "stack_info",
+    "thread",
+    "threadName",
+}
 
-class TerseJsonFormatter:
-    def __init__(
-        self, *args, include_time: bool = True, metadata: Optional[dict] = None
-    ):
-        self.include_time = include_time
-        self.metadata = metadata or {}
 
+class TerseJsonFormatter(logging.Formatter):
     def format(self, record: logging.LogRecord) -> str:
         event = {
             "log": record.getMessage(),
             "namespace": record.name,
             "level": record.levelname,
+            "time": round(record.created, 2),
         }
 
-        # We want to include the timestamp when forwarding over the network, but
-        # exclude it when we are writing to stdout. This is because the log ingester
-        # (e.g. logstash, fluentd) can add its own timestamp.
-        if self.include_time:
-            event["time"] = round(record.created, 2)
-
-        # Add the metadata information to the event (e.g. the server_name).
-        event.update(self.metadata)
+        # Add any extra attributes to the event.
+        extra_attributes = set(record.__dict__.keys()) - _LOG_RECORD_ATTRIBUTES
```

Comment:
Can we avoid building a set here and instead loop over `record.__dict__.items()`, checking against `_LOG_RECORD_ATTRIBUTES`?

Reply:
That should work, yes. 👍